backend/src/Installer.ts

```diff
@@ -152,71 +160,123 @@ export default class Installer {
     return name + platform
   }
 
-  binaryPath(admin?: boolean) {
+  get extensionName() {
+    let platform = ''
+    if (environment.isWindows32) platform = '.x86-win.exe'
+    else if (environment.isWindows) platform = '.x86_64-64.exe'
+    else if (environment.isMac) platform = '.x86_64-osx'
+    return platform
+  }
+
+  get downloadFileNameMuxer() {
+    return `muxer${this.extensionName}`
+  }
+
+  get downloadFileNameDemuxer() {
+    return `demuxer${this.extensionName}`
+  }
+
+  get downloadFileNameConnectd() {
+    return `connectd${this.extensionName}`
+  }
+
+  binaryPathCLI(admin?: boolean) {
```

Comment:
So the Installer.ts is meant to install a single dependancy. So one installer for each resource. We shouldn't add multi resources to the installer, instead create a new instance of the installer for each resource.